# Player Data Kaggle-Fjelstul

## Objectives

Develop a dataset with Kaggle dataset ID's, Fjestul Dataset IDs, and Player Names.

Thgis will serve as a mathcing tool to assign season data (from the kaggle dataset) to world cup data (from thje Fjestul dataset)


## Acknowledgments

Data in this notebook is taken from Kaggle using Kaggle's API. The dataset in Kaggle is made from data originally from Transfermarket.com.

## Libraries

In [241]:
library(dplyr)
library(here)
library(stringr)
library(stringi)


## Player data (Kaggle)

### Call Data with our Kaggle API

In [242]:
system("kaggle datasets download davidcariboo/player-scores -p kaggle_data")

unzip("kaggle_data/player-scores.zip",exdir = "kaggle_data")


In [243]:
player_data <- read.csv(here("1.DataCleaning-R", "kaggle_data", "players.csv"), stringsAsFactors = FALSE, encoding = "UTF-8")

colnames(player_data)

[1] "player_id"                           
 [2] "first_name"                          
 [3] "last_name"                           
 [4] "name"                                
 [5] "last_season"                         
 [6] "current_club_id"                     
 [7] "player_code"                         
 [8] "country_of_birth"                    
 [9] "city_of_birth"                       
[10] "country_of_citizenship"              
[11] "date_of_birth"                       
[12] "sub_position"                        
[13] "position"                            
[14] "foot"                                
[15] "height_in_cm"                        
[16] "contract_expiration_date"            
[17] "agent_name"                          
[18] "image_url"                           
[19] "international_caps"                  
[20] "international_goals"                 
[21] "current_national_team_id"            
[22] "url"                                 
[23] "current_club_domestic_competition_id"
[24] "current_club_name"                   
[25] "market_value_in_eur"                 
[26] "highest_market_value_in_eur"

In [244]:
player_names <- player_data %>%
    select(player_id, name)



## Clean Data

### Lets get the player names and IDs from Fjestul

In [245]:
worldcup_players <- readRDS(here("1.DataCleaning-R", "Data", "RDS", "FullRoster.rds"))

worldcup_players <- worldcup_players %>%
    select(player_id, full_name) %>%
        distinct(player_id, .keep_all = TRUE)


Since the dataset for worldcup_players and player_namess come from sepearate orders their player_id system is different. It is also likely that some names may not match exaclty, lets investigate this possibility. 

In [246]:
length(setdiff(worldcup_players$full_name, player_names$name))
length(worldcup_players$full_name)
setdiff(worldcup_players$full_name, player_names$name)

[1] 856

[1] 2379

[1] "Loun<U+00E8>s Gaouaoui"     "Madjid Bougherra"          
  [3] "Nadir Belhadj"              "Antar Yahia"               
  [5] "Yazid Mansouri"             "Rafik Sa<U+00EF>fi"        
  [7] "Habib Bella<U+00EF>d"       "Abdelkader La<U+00EF>faoui"
  [9] "Karim Ziani"                "Faouzi Chaouchi"           
 [11] "Adl<U+00E8>ne Guedioura"    "Diego Pozo"                
 [13] "Clemente Rodr<U+00ED>guez"  "Mario Bolatti"             
 [15] "Gabriel Heinze"             "Juan Sebasti<U+00E1>n Ver<U+00F3>n"
 [17] "Ariel Garc<U+00E9>"         "Mart<U+00ED>n Palermo"     
 [19] "Maxi Rodr<U+00ED>guez"      "Lucas Neill"               
 [21] "Tim Cahill"                 "Jason Culina"              
 [23] "Michael Beauchamp"          "Brett Emerton"             
 [25] "Joshua Kennedy"             "Harry Kewell"              
 [27] "Scott Chipperfield"         "Vince Grella"              
 [29] "Eugene Galekovic"           "Richard Garcia"            
 [31] "David Carney"               "Dario Vido<U+0161>i<U+0107>"
 [33] "Mark Bresciano"             "not applicable Maicon"     
 [35] "not applicable L<U+00FA>cio" "not applicable Juan"       
 [37] "not applicable Elano"       "Gilberto Silva"            
 [39] "Lu<U+00ED>s Fabiano"        "not applicable Kak<U+00E1>"
 [41] "not applicable Robinho"     "not applicable Luis<U+00E3>o"
 [43] "not applicable Gilberto"    "not applicable Josu<U+00E9>"
 [45] "not applicable Ramires"     "not applicable Kl<U+00E9>berson"
 [47] "not applicable Nilmar"      "not applicable Doni"       
 [49] "not applicable Grafite"     "Nicolas Nkoulou"           
 [51] "Rigobert Song"              "not applicable Geremi"     
 [53] "Achille Eman<U+00E1>"       "Eric Maxim Choupo-Moting"  
 [55] "Souleymanou Hamidou"        "Mohammadou Idrissou"       
 [57] "Jo<U+00EB>l Matip"          "Guy N'dy Assemb<U+00E9>"   
 [59] "Ismael Fuentes"             "Waldo Ponce"               
 [61] "Humberto Suazo"             "Jorge Valdivia"            
 [63] "Miguel Pinto"               "Gonzalo Fierro"            
 [65] "Rodrigo Millar"             "Esteban Paredes"           
 [67] "Luis Mar<U+00ED>n"          "Thomas S<U+00F8>rensen"    
 [69] "Jesper Gr<U+00F8>nkj<U+00E6>r" "Jon Dahl Tomasson"         
 [71] "Martin J<U+00F8>rgensen"    "Per Kr<U+00F8>ldrup"       
 [73] "S<U+00F8>ren Larsen"        "David James"               
 [75] "Robert Green"               "Stephen Warnock"           
 [77] "Ledley King"                "Emile Heskey"              
 [79] "Eric Abidal"                "Thierry Henry"             
 [81] "Arne Friedrich"             "not applicable Cacau"      
 [83] "Hans-J<U+00F6>rg Butt"      "Daniel Agyei"              
 [85] "Hans Sarpei"                "John Paintsil"             
 [87] "Stephen Appiah"             "Prince Tagoe"              
 [89] "Isaac Vorsah"               "Stephen Ahorlu"            
 [91] "Lee Addy"                   "Richard Kingson"           
 [93] "Kostas Chalkias"            "Giourkas Seitaridis"       
 [95] "Nikos Spyropoulos"          "Angelos Charisteas"        
 [97] "Giorgos Karagounis"         "Dimitris Salpingidis"      
 [99] "Vasilis Torosidis"          "Sotirios Kyrgiakos"        
[101] "Kostas Katsouranis"         "Sakis Prittas"             
[103] "Ricardo Canales"            "Osman Ch<U+00E1>vez"       
[105] "Johnny Palacios"            "V<U+00ED>ctor Bern<U+00E1>rdez"
[107] "Hendry Thomas"              "Ram<U+00F3>n N<U+00FA><U+00F1>ez"
[109] "Carlos Pav<U+00F3>n"        "Jerry Palacios"            
[111] "David Suazo"                "Georgie Welcome"           
[113] "Roger Espinoza"             "Boniek Garc<U+00ED>a"      
[115] "Walter Mart<U+00ED>nez"     "Mauricio Sabill<U+00F3>n"  
[117] "<U+00C9>dgar <U+00C1>lvarez" "Noel Valladares"           
[119] "Danilo Turcios"             "Amado Guevara"             
[121] "Donis Escober"              "Sergio Mendoza"            
[123] "Fabio Cannavaro"            "Gennaro Gattuso"           
[1

We can see a very significant amount of players have mismatched names. 

In [247]:
set.seed(12)

player_names %>%
    slice_sample(n = 20) %>%
        select(name)

name
<chr>
Brad Halliday
Pedro Botelho
Gabriel Palmero
Martin Jiranek
You-hyeon Lee
Islam Issa
Andu Kelati
Riccardo Brosco
Mohammed Abera


After manually inspecting I see some players are just spelled slightly differently in one dataset vs the other. For example Antar/Anthar Yahiya. Let's trouble shoot other issues first since this might require a lot of manual comparison which will be easier to do with less observations.

In [248]:
player_names %>%
  filter(grepl("Anthar Yahia", name)) %>%
    select(name)

worldcup_players %>%
  filter(grepl("Antar Yahia", full_name)) %>%
    select(full_name)

name
<chr>
Anthar Yahia


full_name
<chr>
Antar Yahia


Some of the player names also differ in the datasets because of accents. For example Ronald Araujo.

In [249]:
player_names %>%
  filter(grepl("Ronald Araujo", name)) %>%
    select(name)

worldcup_players %>%
  filter(grepl("Ronald Ara", full_name)) %>%
    select(full_name)

name
<chr>
Ronald Araujo


full_name
<chr>
Ronald Ara<U+00FA>jo


The accents seem to be a relevant issue across a lot of players. Let's write a function to normalize all text by removing accents. Also making sure to decode all text in the process.

In [250]:
# This was done with significant help from Codex AI
decode_unicode_tags <- function(x) {
  fix_one <- function(s) {
    matches <- regmatches(
      s,
      gregexpr("<U\\+[0-9A-Fa-f]{4,6}>", s, perl = TRUE)
    )[[1]]

    if (length(matches) == 1 && matches[1] == "") return(s)

    for (m in matches) {
      hex <- sub("<U\\+([0-9A-Fa-f]{4,6})>", "\\1", m, perl = TRUE)
      replacement <- intToUtf8(strtoi(hex, base = 16))
      s <- sub(m, replacement, s, fixed = TRUE)
    }

    s
  }

  vapply(x, fix_one, character(1))
}

remove_accents <- function(x) {
  x <- decode_unicode_tags(x)
  x <- enc2utf8(x)
  stringi::stri_trans_general(x, "Latin-ASCII")
}

#Test
remove_accents("Ronald Araújo")
remove_accents("Ángel Di María")
remove_accents("Cristiano Ronaldo")

[1] "Ronald Araujo"

[1] "Angel Di Maria"

[1] "Cristiano Ronaldo"

Seems to work pretty well, lets apply it to both of our datasets and then compare again

In [251]:
player_names <- player_names %>%
    mutate(name = remove_accents(name))

worldcup_players <- worldcup_players %>%
    mutate(full_name = remove_accents(full_name))

length(setdiff(worldcup_players$full_name, player_names$name))
length(worldcup_players$full_name)
setdiff(worldcup_players$full_name, player_names$name)

[1] 691

[1] 2379

[1] "Lounes Gaouaoui"            "Madjid Bougherra"          
  [3] "Nadir Belhadj"              "Antar Yahia"               
  [5] "Yazid Mansouri"             "Rafik Saifi"               
  [7] "Habib Bellaid"              "Abdelkader Laifaoui"       
  [9] "Karim Ziani"                "Faouzi Chaouchi"           
 [11] "Diego Pozo"                 "Clemente Rodriguez"        
 [13] "Mario Bolatti"              "Gabriel Heinze"            
 [15] "Juan Sebastian Veron"       "Ariel Garce"               
 [17] "Martin Palermo"             "Maxi Rodriguez"            
 [19] "Lucas Neill"                "Tim Cahill"                
 [21] "Jason Culina"               "Michael Beauchamp"         
 [23] "Brett Emerton"              "Joshua Kennedy"            
 [25] "Harry Kewell"               "Scott Chipperfield"        
 [27] "Vince Grella"               "Eugene Galekovic"          
 [29] "Richard Garcia"             "David Carney"              
 [31] "Dario Vidosic"              "Mark Bresciano"            
 [33] "not applicable Maicon"      "not applicable Lucio"      
 [35] "not applicable Juan"        "not applicable Elano"      
 [37] "Gilberto Silva"             "Luis Fabiano"              
 [39] "not applicable Kaka"        "not applicable Robinho"    
 [41] "not applicable Luisao"      "not applicable Gilberto"   
 [43] "not applicable Josue"       "not applicable Ramires"    
 [45] "not applicable Kleberson"   "not applicable Nilmar"     
 [47] "not applicable Doni"        "not applicable Grafite"    
 [49] "Nicolas Nkoulou"            "Rigobert Song"             
 [51] "not applicable Geremi"      "Achille Emana"             
 [53] "Eric Maxim Choupo-Moting"   "Souleymanou Hamidou"       
 [55] "Mohammadou Idrissou"        "Guy N'dy Assembe"          
 [57] "Ismael Fuentes"             "Waldo Ponce"               
 [59] "Humberto Suazo"             "Jorge Valdivia"            
 [61] "Miguel Pinto"               "Gonzalo Fierro"            
 [63] "Rodrigo Millar"             "Esteban Paredes"           
 [65] "Luis Marin"                 "Jesper Gronkjaer"          
 [67] "Jon Dahl Tomasson"          "David James"               
 [69] "Robert Green"               "Stephen Warnock"           
 [71] "Ledley King"                "Emile Heskey"              
 [73] "Thierry Henry"              "Arne Friedrich"            
 [75] "not applicable Cacau"       "Hans-Jorg Butt"            
 [77] "Daniel Agyei"               "Hans Sarpei"               
 [79] "John Paintsil"              "Stephen Appiah"            
 [81] "Prince Tagoe"               "Isaac Vorsah"              
 [83] "Stephen Ahorlu"             "Lee Addy"                  
 [85] "Richard Kingson"            "Kostas Chalkias"           
 [87] "Giourkas Seitaridis"        "Nikos Spyropoulos"         
 [89] "Angelos Charisteas"         "Giorgos Karagounis"        
 [91] "Dimitris Salpingidis"       "Vasilis Torosidis"         
 [93] "Sotirios Kyrgiakos"         "Kostas Katsouranis"        
 [95] "Sakis Prittas"              "Ricardo Canales"           
 [97] "Osman Chavez"               "Johnny Palacios"           
 [99] "Victor Bernardez"           "Hendry Thomas"             
[101] "Ramon Nunez"                "Carlos Pavon"              
[103] "Jerry Palacios"             "David Suazo"               
[105] "Georgie Welcome"            "Boniek Garcia"             
[107] "Walter Martinez"            "Mauricio Sabillon"         
[109] "Edgar Alvarez"              "Noel Valladares"           
[111] "Danilo Turcios"             "Amado Guevara"             
[113] "Donis Escober"              "Sergio Mendoza"            
[115] "Fabio Cannavaro"            "Gennaro Gattuso"           
[117] "Mauro Camoranesi"           "Gianluca Zambrotta"        
[119] "Boubacar Barry"             "Cheick Tiote"              
[121] "not applicable Gervinho"    "not applicable Romaric"    
[123] "Aruna Dindane"              "Aristide Zogbo"            
[125] "Abdul Kader Keita"         

Our set difference went down 165 observations! Lets keep trouble shooting.

There seems to be a few players with "not applicable" as their first name. Let's explore this issue.

In [252]:
worldcup_players %>%
    filter(grepl("not applicable", full_name)) %>%
        print(n=5)

worldcup_players %>%
    filter(grepl("not applicable", full_name)) %>%
        distinct(full_name) %>%
            nrow()

# A tibble: 88 x 2
  player_id full_name            
  <chr>     <chr>                
1 P-94551   not applicable Maicon
2 P-42918   not applicable Lucio 
3 P-98607   not applicable Juan  
4 P-78264   not applicable Elano 
5 P-58388   not applicable Kaka  
# i 83 more rows


[1] 85

In [253]:
readRDS(here("1.DataCleaning-R", "Data", "RDS", "FullRoster.rds")) %>%
    filter(given_name== "not applicable") %>%
        print(n=10)

# A tibble: 113 x 9
   tournament_id team_name team_id team_code player_id family_name   given_name 
   <chr>         <chr>     <chr>   <chr>     <chr>     <chr>         <chr>      
 1 WC-2010       Brazil    T-09    BRA       P-94551   "Maicon"      not applic~
 2 WC-2010       Brazil    T-09    BRA       P-42918   "L\u00facio"  not applic~
 3 WC-2010       Brazil    T-09    BRA       P-98607   "Juan"        not applic~
 4 WC-2010       Brazil    T-09    BRA       P-78264   "Elano"       not applic~
 5 WC-2010       Brazil    T-09    BRA       P-58388   "Kak\u00e1"   not applic~
 6 WC-2010       Brazil    T-09    BRA       P-13354   "Robinho"     not applic~
 7 WC-2010       Brazil    T-09    BRA       P-32968   "Luis\u00e3o" not applic~
 8 WC-2010       Brazil    T-09    BRA       P-30363   "Gilberto"    not applic~
 9 WC-2010       Brazil    T-09    BRA       P-74066   "Josu\u00e9"  not applic~
10 WC-2010       Brazil    T-09    BRA       P-86499   "Ramires"     not applic~
# i 103 

In [254]:
#Let's check some of these players on the Kaggle dataset
set.seed(123)

player_names %>% 
    select(name) %>%
        filter(grepl("Maicon|Elano|Rodri", name)) %>%
            slice_sample(n=10)

name
<chr>
Ramiro Ruiz Rodriguez
Rodrigo Melro
Fer Rodriguez
Jay Rodriguez
Rodrigo Caseres
Rodrigo Varanda
Diego Rodriguez
Rodrigo Alirio
Rodrigo


Some of the players seem to be only reffered to by one name in both datasets. Lets take out the not applicable and see improvements.

In [255]:
remove_not_applicable <- function(x) {
  gsub("not applicable\\s+", "", x)
}

worldcup_players <- worldcup_players %>%
  mutate(full_name = remove_not_applicable(full_name))

length(setdiff(worldcup_players$full_name, player_names$name))
length(worldcup_players$full_name)
setdiff(worldcup_players$full_name, player_names$name)

[1] 617

[1] 2379

[1] "Lounes Gaouaoui"            "Madjid Bougherra"          
  [3] "Nadir Belhadj"              "Antar Yahia"               
  [5] "Yazid Mansouri"             "Rafik Saifi"               
  [7] "Habib Bellaid"              "Abdelkader Laifaoui"       
  [9] "Karim Ziani"                "Faouzi Chaouchi"           
 [11] "Diego Pozo"                 "Clemente Rodriguez"        
 [13] "Mario Bolatti"              "Gabriel Heinze"            
 [15] "Juan Sebastian Veron"       "Ariel Garce"               
 [17] "Martin Palermo"             "Maxi Rodriguez"            
 [19] "Lucas Neill"                "Tim Cahill"                
 [21] "Jason Culina"               "Michael Beauchamp"         
 [23] "Brett Emerton"              "Joshua Kennedy"            
 [25] "Harry Kewell"               "Scott Chipperfield"        
 [27] "Vince Grella"               "Eugene Galekovic"          
 [29] "Richard Garcia"             "David Carney"              
 [31] "Dario Vidosic"              "Mark Bresciano"            
 [33] "Elano"                      "Gilberto Silva"            
 [35] "Luis Fabiano"               "Kleberson"                 
 [37] "Nilmar"                     "Doni"                      
 [39] "Grafite"                    "Nicolas Nkoulou"           
 [41] "Rigobert Song"              "Geremi"                    
 [43] "Achille Emana"              "Eric Maxim Choupo-Moting"  
 [45] "Souleymanou Hamidou"        "Mohammadou Idrissou"       
 [47] "Guy N'dy Assembe"           "Ismael Fuentes"            
 [49] "Waldo Ponce"                "Humberto Suazo"            
 [51] "Jorge Valdivia"             "Miguel Pinto"              
 [53] "Gonzalo Fierro"             "Rodrigo Millar"            
 [55] "Esteban Paredes"            "Luis Marin"                
 [57] "Jesper Gronkjaer"           "Jon Dahl Tomasson"         
 [59] "David James"                "Robert Green"              
 [61] "Stephen Warnock"            "Ledley King"               
 [63] "Emile Heskey"               "Thierry Henry"             
 [65] "Arne Friedrich"             "Hans-Jorg Butt"            
 [67] "Daniel Agyei"               "Hans Sarpei"               
 [69] "John Paintsil"              "Stephen Appiah"            
 [71] "Prince Tagoe"               "Isaac Vorsah"              
 [73] "Stephen Ahorlu"             "Lee Addy"                  
 [75] "Richard Kingson"            "Kostas Chalkias"           
 [77] "Giourkas Seitaridis"        "Nikos Spyropoulos"         
 [79] "Angelos Charisteas"         "Giorgos Karagounis"        
 [81] "Dimitris Salpingidis"       "Vasilis Torosidis"         
 [83] "Sotirios Kyrgiakos"         "Kostas Katsouranis"        
 [85] "Sakis Prittas"              "Ricardo Canales"           
 [87] "Osman Chavez"               "Johnny Palacios"           
 [89] "Victor Bernardez"           "Hendry Thomas"             
 [91] "Ramon Nunez"                "Carlos Pavon"              
 [93] "Jerry Palacios"             "David Suazo"               
 [95] "Georgie Welcome"            "Boniek Garcia"             
 [97] "Walter Martinez"            "Mauricio Sabillon"         
 [99] "Edgar Alvarez"              "Noel Valladares"           
[101] "Danilo Turcios"             "Amado Guevara"             
[103] "Donis Escober"              "Sergio Mendoza"            
[105] "Fabio Cannavaro"            "Gennaro Gattuso"           
[107] "Mauro Camoranesi"           "Gianluca Zambrotta"        
[109] "Boubacar Barry"             "Cheick Tiote"              
[111] "Aruna Dindane"              "Aristide Zogbo"            
[113] "Abdul Kader Keita"          "Daniel Yeboah"             
[115] "Seigo Narazaki"             "Yuki Abe"                  
[117] "Yuichi Komano"              "Marcus Tulio Tanaka"       
[119] "Yasuhito Endo"              "Daisuke Matsui"            
[121] "Shunsuke Nakamura"          "Keiji Tamada"              
[123] "Kisho Yano"                 "Daiki Iwamasa"             
[125] "Kengo Nakamura"            

Great! 74 out of the 85 players with not applicable were solved with this issue. Lets asses the other 11. We can do this by checking all full names with no spaces in the sett differences list. If we have 11 then we know we got all of the not applicables we're missing.

In [256]:
unmatched_players <- setdiff(worldcup_players$full_name, player_names$name)

one_word_unmatched <- unmatched_players[!grepl("\\s", unmatched_players)]

length(one_word_unmatched)
one_word_unmatched

[1] 11

[1] "Elano"     "Kleberson" "Nilmar"    "Doni"      "Grafite"   "Geremi"   
 [7] "Yakubu"    "Miguel"    "Deco"      "Cedric"    "Ro-Ro"

In [257]:
player_data_clean <- player_data %>%
  mutate(clean_name = remove_not_applicable(remove_accents(name)))

pattern <- paste0("\\b(", paste(one_word_unmatched, collapse = "|"), ")\\b")

player_data_clean %>%
  filter(grepl(pattern, clean_name)) %>%
    filter(!(clean_name %in% worldcup_players$full_name)) %>%
      distinct(name, clean_name, last_season, international_caps) %>%
        filter(last_season>= 2010 & international_caps > 0)



name,clean_name,last_season,international_caps
<chr>,<chr>,<int>,<int>
Miguel Lopes,Miguel Lopes,2024,4
Miguel Lloyd,Miguel Lloyd,2025,47
Luis Miguel Rodr<U+00ED>guez,Luis Miguel Rodriguez,2024,1
C<U+00E9>dric Soares,Cedric Soares,2025,34
C<U+00E9>dric Bakambu,Cedric Bakambu,2025,67
C<U+00E9>dric Hountondji,Cedric Hountondji,2024,29
Cedric Teuchert,Cedric Teuchert,2025,8
Pedro Miguel,Pedro Miguel,2025,99
Cedric Itten,Cedric Itten,2025,13


After further research Im comfortable assigning the full names to the following players.

In [259]:
worldcup_players <- worldcup_players %>%
  mutate(
    full_name = case_when(
      full_name == "Elano" ~ "Elano Yegen",
      full_name == "Cedric" ~ "Cedric Soares",
      full_name == "Ro-Ro" ~ "Pedro Miguel",
      TRUE ~ full_name
    )
  )

The other few players with one word names we will leave as undefined for now.


I will Have CODEX AI go thorugh the full list of differences and through the full p[layers.csv dataset and find clear minimal gramatical mismatches. I willl then manually check a few to minimize likelihood of errors.

In [261]:
# Obvious name fixes found by comparing unmatched World Cup names to players.csv
# Only includes high-confidence spelling, punctuation, or known single-name expansions.
worldcup_players <- worldcup_players %>%
  mutate(
    full_name = case_when(
      full_name == "Abdulrahman Al-Aboud" ~ "Abdulrahman Al-Oboud",
      full_name == "Eric Maxim Choupo-Moting" ~ "Eric-Maxim Choupo-Moting",
      full_name == "Lawrence Ati-Zigi" ~ "Lawrence Ati Zigi",
      full_name == "Abdulaziz Hatem" ~ "Abdulaziz Hatim",
      full_name == "Abdulellah Al-Malki" ~ "Abdulelah Al-Malki",
      full_name == "Ahmed Hegazi" ~ "Ahmed Hegazy",
      full_name == "Aleksandr Yerokhin" ~ "Aleksandr Erokhin",
      full_name == "Aleksei Ionov" ~ "Aleksey Ionov",
      full_name == "Aleksei Kozlov" ~ "Aleksey Kozlov",
      full_name == "Aleksei Miranchuk" ~ "Aleksey Miranchuk",
      full_name == "Andrey Yeshchenko" ~ "Andrey Eshchenko",
      full_name == "Antar Yahia" ~ "Anthar Yahia",
      full_name == "Bilal El Khannous" ~ "Bilal El Khannouss",
      full_name == "Brian Idowu" ~ "Bryan Idowu",
      full_name == "Cheick Tiote" ~ "Cheik Tiote",
      full_name == "Daler Kuzyayev" ~ "Daler Kuzyaev",
      full_name == "Dimitris Salpingidis" ~ "Dimitrios Salpingidis",
      full_name == "Dmitri Kombarov" ~ "Dmitriy Kombarov",
      full_name == "Erick Davis" ~ "Eric Davis",
      full_name == "Filip Duricic" ~ "Filip Djuricic",
      full_name == "Georgi Shchennikov" ~ "Georgiy Shchennikov",
      full_name == "Giorgian De Arrascaeta" ~ "Giorgian de Arrascaeta",
      full_name == "Guy N'dy Assembe" ~ "Guy Ndy Assembe",
      full_name == "Hannes THor Halldorsson" ~ "Hannes Thor Halldorsson",
      full_name == "Hassan Al-Haydos" ~ "Hasan Al-Haydos",
      full_name == "Marco Urena" ~ "Marcos Urena",
      full_name == "Mateus Uribe" ~ "Matheus Uribe",
      full_name == "Michael Babatunde" ~ "Michel Babatunde",
      full_name == "Sergei Ignashevich" ~ "Sergey Ignashevich",
      full_name == "Sotirios Kyrgiakos" ~ "Sotiris Kyrgiakos",
      full_name == "Srdan Babic" ~ "Srdjan Babic",
      full_name == "Steve Cherundolo" ~ "Steven Cherundolo",
      full_name == "Vasili Berezutski" ~ "Vasiliy Berezutski",
      full_name == "Vasilis Torosidis" ~ "Vasilios Torosidis",
      full_name == "Viktor Fayzulin" ~ "Viktor Faizulin",
      full_name == "William Pacho" ~ "Willian Pacho",
      full_name == "Yuri Lodygin" ~ "Yuriy Lodygin",
      full_name == "Yuri Zhirkov" ~ "Yuriy Zhirkov",
      full_name == "Andrey Lunyov" ~ "Andrey Lunev",
      full_name == "Dani Carvajal" ~ "Daniel Carvajal",
      full_name == "Fyodor Kudryashov" ~ "Fedor Kudryashov",
      full_name == "Fyodor Smolov" ~ "Fedor Smolov",
      full_name == "Georges-Kevin Nkoudou" ~ "Georges-Kevin N'Koudou",
      full_name == "Giorgos Karagounis" ~ "Georgios Karagounis",
      full_name == "Giorgos Tzavellas" ~ "Georgios Tzavellas",
      full_name == "John Heitinga" ~ "Johnny Heitinga",
      full_name == "Joo-ho Park" ~ "Ju-ho Park",
      full_name == "Mbark Boussoufa" ~ "Moubarak Boussoufa",
      full_name == "Nikos Spyropoulos" ~ "Nikolaos Spyropoulos",
      full_name == "Robert Green" ~ "Rob Green",
      full_name == "Yakubu" ~ "Yakubu Aiyegbeni",
      TRUE ~ full_name
    )
  )


In [262]:
length(setdiff(worldcup_players$full_name, player_names$name))
length(worldcup_players$full_name)
setdiff(worldcup_players$full_name, player_names$name)

[1] 563

[1] 2379

[1] "Lounes Gaouaoui"            "Madjid Bougherra"          
  [3] "Nadir Belhadj"              "Yazid Mansouri"            
  [5] "Rafik Saifi"                "Habib Bellaid"             
  [7] "Abdelkader Laifaoui"        "Karim Ziani"               
  [9] "Faouzi Chaouchi"            "Diego Pozo"                
 [11] "Clemente Rodriguez"         "Mario Bolatti"             
 [13] "Gabriel Heinze"             "Juan Sebastian Veron"      
 [15] "Ariel Garce"                "Martin Palermo"            
 [17] "Maxi Rodriguez"             "Lucas Neill"               
 [19] "Tim Cahill"                 "Jason Culina"              
 [21] "Michael Beauchamp"          "Brett Emerton"             
 [23] "Joshua Kennedy"             "Harry Kewell"              
 [25] "Scott Chipperfield"         "Vince Grella"              
 [27] "Eugene Galekovic"           "Richard Garcia"            
 [29] "David Carney"               "Dario Vidosic"             
 [31] "Mark Bresciano"             "Gilberto Silva"            
 [33] "Luis Fabiano"               "Kleberson"                 
 [35] "Nilmar"                     "Doni"                      
 [37] "Grafite"                    "Nicolas Nkoulou"           
 [39] "Rigobert Song"              "Geremi"                    
 [41] "Achille Emana"              "Souleymanou Hamidou"       
 [43] "Mohammadou Idrissou"        "Ismael Fuentes"            
 [45] "Waldo Ponce"                "Humberto Suazo"            
 [47] "Jorge Valdivia"             "Miguel Pinto"              
 [49] "Gonzalo Fierro"             "Rodrigo Millar"            
 [51] "Esteban Paredes"            "Luis Marin"                
 [53] "Jesper Gronkjaer"           "Jon Dahl Tomasson"         
 [55] "David James"                "Stephen Warnock"           
 [57] "Ledley King"                "Emile Heskey"              
 [59] "Thierry Henry"              "Arne Friedrich"            
 [61] "Hans-Jorg Butt"             "Daniel Agyei"              
 [63] "Hans Sarpei"                "John Paintsil"             
 [65] "Stephen Appiah"             "Prince Tagoe"              
 [67] "Isaac Vorsah"               "Stephen Ahorlu"            
 [69] "Lee Addy"                   "Richard Kingson"           
 [71] "Kostas Chalkias"            "Giourkas Seitaridis"       
 [73] "Angelos Charisteas"         "Kostas Katsouranis"        
 [75] "Sakis Prittas"              "Ricardo Canales"           
 [77] "Osman Chavez"               "Johnny Palacios"           
 [79] "Victor Bernardez"           "Hendry Thomas"             
 [81] "Ramon Nunez"                "Carlos Pavon"              
 [83] "Jerry Palacios"             "David Suazo"               
 [85] "Georgie Welcome"            "Boniek Garcia"             
 [87] "Walter Martinez"            "Mauricio Sabillon"         
 [89] "Edgar Alvarez"              "Noel Valladares"           
 [91] "Danilo Turcios"             "Amado Guevara"             
 [93] "Donis Escober"              "Sergio Mendoza"            
 [95] "Fabio Cannavaro"            "Gennaro Gattuso"           
 [97] "Mauro Camoranesi"           "Gianluca Zambrotta"        
 [99] "Boubacar Barry"             "Aruna Dindane"             
[101] "Aristide Zogbo"             "Abdul Kader Keita"         
[103] "Daniel Yeboah"              "Seigo Narazaki"            
[105] "Yuki Abe"                   "Yuichi Komano"             
[107] "Marcus Tulio Tanaka"        "Yasuhito Endo"             
[109] "Daisuke Matsui"             "Shunsuke Nakamura"         
[111] "Keiji Tamada"               "Kisho Yano"                
[113] "Daiki Iwamasa"              "Kengo Nakamura"            
[115] "Yasuyuki Konno"             "Yoshito Okubo"             
[117] "Junichi Inamoto"            "Yuji Nakazawa"             
[119] "Yoshikatsu Kawaguchi"       "Oscar Perez"               
[121] "Francisco Javier Rodriguez" "Carlos Salcido"            
[123] "Rafael Marquez"             "Ricardo Osorio"            
[125] "Gerardo Torrado"           

I want to assses the relevance of this unmatched players to our dataset and start considering the possibility of exc;luding some of these players.

In [274]:
nonmatched <- setdiff(worldcup_players$full_name, player_names$name)
fullworldcup_players <- readRDS(here("1.DataCleaning-R", "Data", "RDS", "FullRoster.rds"))

fullworldcup_players %>% 
    filter(full_name %in% nonmatched) %>%
        group_by(team_name, tournament_id) %>%
            summarize(players= sum(n())) %>%
                arrange(desc(players))




`summarise()` has grouped output by 'team_name'. You can override using the
`.groups` argument.


team_name,tournament_id,players
<chr>,<chr>,<int>
North Korea,WC-2010,23
New Zealand,WC-2010,18
South Korea,WC-2022,18
South Africa,WC-2010,17
Iran,WC-2014,15
South Korea,WC-2010,14
Australia,WC-2010,13
Japan,WC-2010,13
United States,WC-2010,12


Looking at this table we can see two clear trends. Firstl;y, some countries have almost all their players missing, there's a big chance that some of this countries leagues are simply not in our dataste and hence most of their national team players arent available. The other trend is that the oldest the tournamnet the most missing players. Since there is a good chance many of this players we just dont have data for I've decided to take all temas with 8 or more missing players out of our training data. 

In [277]:
fullworldcup_players %>% 
  filter(full_name %in% nonmatched) %>%
    group_by(team_name, tournament_id) %>%
        summarize(players = n(), .groups = "drop") %>%
            filter(players >= 8) %>%
                summarize(total_players = sum(players))

563-265

total_players
<int>
265


[1] 298

Now we only have 298 missing players, this is reasonable at around 10% of total players. It's inevitable to work with some incomplete data and assuming most of the missing plkayers are lower profile players we can assume that there effect on team performances would be lower so the model shouldn't be affected too strongly.

In [282]:
matched_players <- worldcup_players %>%
  rename(player_id_fj = player_id) %>%
  distinct(full_name, .keep_all = TRUE) %>%
  inner_join(
    player_names %>%
      rename(
        player_id_kag = player_id,
        full_name = name
      ) %>%
      distinct(full_name, .keep_all = TRUE),
    by = "full_name"
  ) %>%
  select(full_name, player_id_fj, player_id_kag)

In [284]:
str(matched_players)
head(matched_players)
dim(matched_players)
any(duplicated(matched_players$player_id_fj))
any(duplicated(matched_players$player_id_kag))

tibble [1,811 x 3] (S3: tbl_df/tbl/data.frame)
 $ full_name    : chr [1:1811] "Anthar Yahia" "Rafik Halliche" "Ryad Boudebouz" "Mehdi Lacen" ...
 $ player_id_fj : chr [1:1811] "P-77742" "P-19453" "P-70903" "P-23484" ...
 $ player_id_kag: int [1:1811] 5586 67815 77826 35539 76406 20903 19385 74768 8060 12351 ...


full_name,player_id_fj,player_id_kag
<chr>,<chr>,<int>
Anthar Yahia,P-77742,5586
Rafik Halliche,P-19453,67815
Ryad Boudebouz,P-70903,77826
Mehdi Lacen,P-23484,35539
Abdelkader Ghezzal,P-22008,76406
Rafik Djebbour,P-10693,20903


[1] 1811    3

[1] FALSE

[1] FALSE

In [287]:
saveRDS(matched_players, here("1.DataCleaning-R", "Data", "RDS", "Matching.rds"))